File: arcgis_map_android/android/src/main/kotlin/esri/arcgis/flutter_plugin/CenterPositionStreamHandler.kt
package esri.arcgis.flutter_plugin

import esri.arcgis.flutter_plugin.model.LatLng
import io.flutter.plugin.common.EventChannel

class CenterPositionStreamHandler : EventChannel.StreamHandler {
    private var sink: EventChannel.EventSink? = null

    fun add(center: LatLng) {
        sink?.success(toMap(center))
    }

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        sink = events
    }

    override fun onCancel(arguments: Any?) {
        sink = null
    }
}

File: arcgis_map_sdk_ios/ios/arcgis_map_sdk_ios/Sources/arcgis_map_sdk_ios/CenterPositionStreamHandler.swift
//
//  CenterPositionStreamHandler.swift
//  arcgis_map_sdk_ios
//
//  Created by Tarek Tolba on 06/02/2023.
//

import Foundation
import Flutter

class CenterPositionStreamHandler: NSObject, FlutterStreamHandler {

    private var sink: FlutterEventSink?

    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        sink = events
        return nil
    }

    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        sink = nil
        return nil
    }

    func add(center: LatLng) {
        sink?(center.dictionary)
    }
}
